src/Compilers/Core/Portable/PEWriter/MetadataWriter.PortablePdb.cs

```diff
@@ -831,5 +834,100 @@ private void EmbedSourceLink(Stream stream)
                 kind: _debugMetadataOpt.GetOrAddGuid(PortableCustomDebugInfoKinds.SourceLink),
                 value: _debugMetadataOpt.GetOrAddBlob(bytes));
         }
+
+        /// <summary>
+        /// Capture the set of compilation options to allow a compilation 
+        /// to be reconstructed from the pdb
+        /// </summary>
+        private void EmbedCompilationOptions(CommonPEModuleBuilder module)
+        {
+
+            var builder = new BlobBuilder();
+
+            // compilerversion
+            var compilerVersion = typeof(Compilation).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion;
+            WriteValue("compilerversion", compilerVersion);
```

Comment:
Where is the compiler SHA being written?

Reply:
The SHA is included in the informational version (the format is `x.y.z-label+sha`)

Reply:
Is it worth adding to the unit tests that the compiler version follows a specific format? Just to make sure the information embedded in the pdb is what we want

Reply:
I'm not sure it's included there at all times. Locally testing this is the assemblyinfo generated

```csharp
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by a tool.
//     Runtime Version:4.0.30319.42000
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

using System;
using System.Reflection;

[assembly: System.Reflection.AssemblyCompanyAttribute("Microsoft Corporation")]
[assembly: System.Reflection.AssemblyConfigurationAttribute("Debug")]
[assembly: System.Reflection.AssemblyCopyrightAttribute("© Microsoft Corporation. All rights reserved.")]
[assembly: System.Reflection.AssemblyFileVersionAttribute("42.42.42.42424")]
[assembly: System.Reflection.AssemblyInformationalVersionAttribute("3.7.0-dev")]
[assembly: System.Reflection.AssemblyProductAttribute("BuildBoss")]
[assembly: System.Reflection.AssemblyTitleAttribute("BuildBoss")]
[assembly: System.Reflection.AssemblyVersionAttribute("42.42.42.42")]
[assembly: System.Resources.NeutralResourcesLanguageAttribute("en-US")]

// Generated by the MSBuild WriteCodeFragment class.
```

`[assembly: System.Reflection.AssemblyInformationalVersionAttribute("3.7.0-dev")]` 

that makes the string `3.7.0-dev`

Probably does different for official builds? Is there a good way to add a test for this?

Reply:
Yes, official builds will always have it. Commit sha is not useful for local builds.

Reply:
Re testing - I think the best we can do is to use `Microsoft.CodeAnalysis.CommitHashAttribute`.

Add
```xml
    <GenerateMicrosoftCodeAnalysisCommitHashAttribute>true</GenerateMicrosoftCodeAnalysisCommitHashAttribute>
```
to the csproj that contains the test (if it's not there yet) and then check the value of the attribute. If it's `<developer build>` then it's dev build that doesn't have the commit sha, other wise it contains commit sha that should be the same as the one in informational version.